### ISSUE TITLE
Can't close\kill\fullsize applications using shortcut after suspend and restore

### ISSUE THREAD

**Issue by forGGe** [awesomeWM/awesome#659]
Hi everyone! I hit strange bug recently after (probably) upgrading my awesome wm. After suspending and then restoring my session I can't operate over already running application using Awesome hotkeys.

Environment:
Awesome v. 3.5.7-1 , fresh rc.lua
Xorg-server v. 1.18.0-4
```
uname -a
Linux Host 4.3.3-3-ARCH #1 SMP PREEMPT Wed Jan 20 08:12:23 CET 2016 x86_64 GNU/Linux
```

Steps to reproduce:
 - open any amount of applications in awesome wm
 - make sure your Mod+Shift+C combination closes a window
 - suspend your PC
 - restore PC
 - try to close a window using shortcut, it is no longer works.
 - few more hotkeys doesn't work either (like minimizing, changing geometry, etc.)

More fun:
 - open new application
 - you can now close its window using Mod+Shift+C

Here is what xev shows me, when I'm trying to kill it and it still lives:
```
KeyPress event, serial 35, synthetic NO, window 0x2600001,
    root 0xd5, subw 0x0, time 56186826, (164,47), root:(1031,717),
    state 0x0, keycode 133 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyPress event, serial 35, synthetic NO, window 0x2600001,
    root 0xd5, subw 0x0, time 56187430, (164,47), root:(1031,717),
    state 0x40, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

FocusOut event, serial 35, synthetic NO, window 0x2600001,
    mode NotifyGrab, detail NotifyAncestor

FocusOut event, serial 35, synthetic NO, window 0x2600001,
    mode NotifyUngrab, detail NotifyPointer

FocusIn event, serial 35, synthetic NO, window 0x2600001,
    mode NotifyUngrab, detail NotifyAncestor

KeymapNotify event, serial 35, synthetic NO, window 0x0,
    keys:  2   0   0   0   0   0   4   0   0   0   0   0   0   0   0   0   
           32  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

KeyRelease event, serial 35, synthetic NO, window 0x2600001,
    root 0xd5, subw 0x0, time 56192388, (164,47), root:(1031,717),
    state 0x41, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyRelease event, serial 35, synthetic NO, window 0x2600001,
    root 0xd5, subw 0x0, time 56192798, (164,47), root:(1031,717),
    state 0x40, keycode 133 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XFilterEvent returns: False

```

In contrast, freshly launched xev can be closed and shows this log:
```
KeyPress event, serial 33, synthetic NO, window 0x2000001,
    root 0xd5, subw 0x0, time 56258125, (76,124), root:(1623,144),
    state 0x0, keycode 133 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyPress event, serial 33, synthetic NO, window 0x2000001,
    root 0xd5, subw 0x0, time 56258303, (76,124), root:(1623,144),
    state 0x40, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

ClientMessage event, serial 33, synthetic YES, window 0x2000001,
    message_type 0x18e (WM_PROTOCOLS), format 32, message 0x18f (WM_DELETE_WINDOW)
```

Please let me know if more information is required, I will gladly post it here.

**Comment by fgro** [comment]
+1 .. this is 3.5.7 on Gentoo. I have similar behaviour, though it seems that I don't need to suspend to get the malfunction but rather after a certain time the window was opened. So it's hard for me to reproduce in a deterministic way (how much time needs to pass for the malfunction to occur? is it maybe some other application in the system that interferes?)

**Comment by eyqs** [comment]
Completely vanilla awesome 3.5.7-1 on my 4.3.3-3-ARCH box confirming, though freezes only happen after suspending and not after a long time (maybe I'm not waiting long enough).

Closing (Mod+Shift+C), minimizing (Mod+N), setting fullscreen (Mod+F), and sending to next screen (Mod+O) never work on all previously opened windows after suspend. Mod+H and Mod+L sometimes work and sometimes don't. With chromium open, if I add a new window in the same tag, Mod+H and Mod+L don't do anything; but if I move chromium into a tag that already has windows in it, then Mod+H and Mod+L work. What a peculiar problem...

**Comment by subbyte** [comment]
+1 for Awesome 3.5.7-1 on Archlinux. It seems that Mod4 does not fire. I cannot close my Firefox, Chromium and even a terminal after a while. When I press Mod4 + Shift + c in the terminal (Mod4 is Alt in my settings), I get a capital letter "C". However, Mod4 actually works, and I can switch to another tag without any problem.

**Comment by sim590** [comment]
+1 on archlinux too. I experience it the same way as fgro.

**Comment by thibautf** [comment]
+ 1
Archlinux + awesome 3.5.7-1

**Comment by psychon** [comment]
Fixed in 3.5.8 (this fix is the only reason that 3.5.8 exists). The problem occurs when the keyboard layout is changed (however, no idea why the X server tells awesome that the keyboard layout changed after suspend/resume).
